### PR TITLE
Upgrade to version3

### PR DESCRIPTION
**Pull request by Sanket-0510**
I have successfully added the support for version 3, Right now the templates are updated and I am working over updating the test for version 3. Will update the tests in the next commit and it will be ready for merging.

### CODE REVIEW

File: samples/fablo-config-hlf3-1orgs-1chaincode.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      {
+        "organization": {
+          "name": "Orderer",
+          "domain": "orderer.example.com"
+        },
+        "orderers": [
+          {
+            "groupName": "group1",
+            "type": "raft",
+            "instances": 1
```
Comment (dzikowski) [inline samples/fablo-config-hlf3-1orgs-1chaincode.json:20]: please add at least 3 raft nodes, and also verify them in logs (we need to be sure all raft nodes are able to support the channel)

File: src/setup-docker/index.ts [hunk 2]
```diff
@@ -213,20 +213,15 @@ export default class SetupDockerGenerator extends Generator {
   }
 
   _copyUtilityScripts(capabilities: Capabilities): void {
-    this.fs.copyTpl(this.templatePath("fabric-docker.sh"), this.destinationPath("fabric-docker.sh"));
+    this.fs.copyTpl(this.templatePath(`fabric-docker.sh`), this.destinationPath("fabric-docker.sh"));
```
Comment (dzikowski) [inline src/setup-docker/index.ts:216]:
```suggestion
    this.fs.copyTpl(this.templatePath("fabric-docker.sh"), this.destinationPath("fabric-docker.sh"));
```

File: src/setup-docker/templates/fabric-docker/.env [hunk 3]
```diff
@@ -17,9 +17,12 @@ FABRIC_CA_VERSION=<%= global.fabricCaVersion %>
...
 FABRIC_CCENV_VERSION=<%= global.fabricCcenvVersion %>
 FABRIC_BASEOS_VERSION=<%= global.fabricBaseosVersion %>
-FABRIC_JAVAENV_VERSION=<%= global.fabricJavaenvVersion %>
-FABRIC_NODEENV_VERSION=<%= global.fabricNodeenvVersion %>
-RECOMMENDED_NODE_VERSION=<%= global.fabricRecommendedNodeVersion %>
+
+
+FABRIC_JAVAENV_VERSION=3.0-beta
+FABRIC_NODEENV_VERSION=2.5
+RECOMMENDED_NODE_VERSION=18
```
Comment (dzikowski) [inline src/setup-docker/templates/fabric-docker/.env:24]: should be taken from `global.`

File: src/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 4]
```diff
@@ -41,22 +46,54 @@ generateChannelsArtifacts() {
...
+              docker exec -i <%= org.cli.address %> bash -c <% -%>
+                "source scripts/channel_fns.sh; fetchChannelAndJoin '<%= channel.name %>' '<%= org.mspName %>' '<%= peer.fullAddress %>' 'crypto/users/Admin@<%= org.domain %>/msp' '<%= channel.ordererHead.fullAddress %>';"
+            <% } else { -%>
+              docker exec -i <%= org.cli.address %> bash -c <% -%>
+                "source scripts/channel_fns.sh; fetchChannelAndJoinTls '<%= channel.name %>' '<%= org.mspName %>' '<%= peer.fullAddress %>' 'crypto/users/Admin@<%= org.domain %>/msp' 'crypto/users/Admin@<%= org.domain %>/tls' 'crypto-orderer/tlsca.<%= channel.ordererHead.domain %>-cert.pem' '<%= channel.ordererHead.fullAddress %>';"
+            <% } -%>
+          <% } -%>
+        <% }) -%>
+      <% }) -%>
+    <% }) -%>
```
Comment (dzikowski) [inline src/setup-docker/templates/fabric-docker/commands-generated.sh:80]: I believe it misses our changes with multiple orderers

File: src/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 5]
```diff
@@ -70,20 +107,19 @@ installChannels() {
...
     <% chaincodes.forEach((chaincode) => { -%>
       if [ -n "$(ls "$CHAINCODES_BASE_DIR/<%= chaincode.directory %>")" ]; then
-        <% if (global.capabilities.isV2) { -%>
           <% if (global.peerDevMode) { -%>
             <%- include('commands-generated/chaincode-dev-v2.sh', { chaincode }); -%>
           <% } else { -%>
             local version="<%= chaincode.version %>"
             <%- include('commands-generated/chaincode-install-v2.sh', { chaincode, global }); -%>
           <% } -%>
-        <% } -%>
```
Comment (dzikowski) [inline src/setup-docker/templates/fabric-docker/commands-generated.sh:86]: you can decrease identation

File: src/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 6]
```diff
@@ -154,6 +184,7 @@ upgradeChaincode() {
 }
 
 notifyOrgsAboutChannels() {
+  <% if(!global.capabilities.isV3) { %>
   printHeadline "Creating new channel config blocks" "U1F537"
   <% channels.forEach((channel) => { -%>
     <% channel.orgs.forEach((org) => { -%>
```
Comment (dzikowski) [inline src/setup-docker/templates/fabric-docker/commands-generated.sh:190]: please keep consistent indentation (increase)

File: src/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 7]
```diff
@@ -194,6 +225,11 @@ notifyOrgsAboutChannels() {
       deleteNewChannelUpdateTx "<%= channel.name %>" "<%= org.mspName %>" "<%= org.cli.address %>"
     <% }) -%>
   <% }) -%>
+
+  <% } else { %> 
+  echo ""
+  <% } %>
```
Comment (dzikowski) [inline src/setup-docker/templates/fabric-docker/commands-generated.sh:231]: indentation

File: src/setup-docker/templates/fabric-docker/scripts/base-functions-v2.sh [hunk 8]
```diff
@@ -95,10 +95,12 @@ createChannelTx() {
   docker cp "$CONFIG_PATH" $CONTAINER_NAME:/fabric-config || removeContainer $CONTAINER_NAME
 
   docker exec -i $CONTAINER_NAME mkdir /config || removeContainer $CONTAINER_NAME
-  docker exec -i $CONTAINER_NAME configtxgen --configPath ./fabric-config -profile "${CONFIG_PROFILE}" -outputCreateChannelTx ./config/channel.tx -channelID "${CHANNEL_NAME}" || removeContainer $CONTAINER_NAME
 
+
```
Comment (dzikowski) [inline src/setup-docker/templates/fabric-docker/scripts/base-functions-v2.sh:99]: please remove any duplicated lines

File: src/setup-docker/templates/fabric-docker/commands-generated.sh (hunk 4 above)
Comment (Sanket-0510) [reply src/setup-docker/templates/fabric-docker/commands-generated.sh]: Yeah I merged the changes

File: e2e-network/docker/test-05-version3-BFT.sh [hunk 9]
```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+set -eux
```
Comment (dzikowski) [inline e2e-network/docker/test-05-version3-BFT.sh:3]:
```suggestion
set -eu
```

File: e2e-network/docker/test-05-version3.sh [hunk 10]
```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+set -eux
```
Comment (dzikowski) [inline e2e-network/docker/test-05-version3.sh:3]:
```suggestion
set -eu
```

File: e2e/__snapshots__/fablo-config-hlf2-1org-1chaincode-raft-explorer.json.test.ts.snap [hunk 11]
```diff
@@ -1715,6 +1718,7 @@ generateChannelsArtifacts() {
 }
 
 installChannels() {
+  set -x
```
Comment (dzikowski) [inline e2e/__snapshots__/fablo-config-hlf2-1org-1chaincode-raft-explorer.json.test.ts.snap:1721]: to be removed

File: src/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 12]
```diff
@@ -39,24 +42,60 @@ generateChannelsArtifacts() {
 }
 
 installChannels() {
+  set -x
```
Comment (dzikowski) [inline src/setup-docker/templates/fabric-docker/commands-generated.sh:45]: to be removed

File: e2e/__snapshots__/fablo-config-hlf2-1org-1chaincode-raft-explorer.json.test.ts.snap (hunk 11 above)
Comment (Sanket-0510) [reply e2e/__snapshots__/fablo-config-hlf2-1org-1chaincode-raft-explorer.json.test.ts.snap:1721]: done

File: src/setup-docker/templates/fabric-docker/commands-generated.sh (hunk 12 above)
Comment (Sanket-0510) [reply src/setup-docker/templates/fabric-docker/commands-generated.sh:45]: done

**Pull request closed without merging**